Include/cpython/object.h

```diff
@@ -296,7 +302,8 @@ typedef struct _heaptypeobject {
 PyAPI_FUNC(const char *) _PyType_Name(PyTypeObject *);
 PyAPI_FUNC(PyObject *) _PyType_Lookup(PyTypeObject *, PyObject *);
 PyAPI_FUNC(PyObject *) _PyType_LookupId(PyTypeObject *, _Py_Identifier *);
-PyAPI_FUNC(PyObject *) _PyObject_LookupSpecial(PyObject *, _Py_Identifier *);
+PyAPI_FUNC(PyObject *) _PyObject_LookupSpecial(PyObject *, PyObject *);
```

Comment:
I know the name of this function should have `Id` on the end, but I don't think we should be breaking the API.
Maybe drop the leading underscore on the new version:
```C
/* Legacy API -- Use PyObject_LookupSpecial instead */
PyAPI_FUNC(PyObject *) _PyObject_LookupSpecial(PyObject *, _Py_Identifier *);

PyAPI_FUNC(PyObject *) PyObject_LookupSpecial(PyObject *, PyObject *);
```


I'm surprised that there is no "lookup special" function in the C-API, it seems like a fundamental operation.